Zend/zend.c

```diff
@@ -66,6 +66,15 @@ static uint32_t zend_version_info_length;
 #define ZEND_CORE_VERSION_INFO	"Zend Engine v" ZEND_VERSION ", Copyright (c) Zend Technologies\n"
 #define PRINT_ZVAL_INDENT 4
 
+#ifdef HAVE_SIGPROCMASK
+/* for blocking/unblocking signals */
+ZEND_API sigset_t mask_all_signals;
+
+# if ZEND_DEBUG
+ZEND_API TSRM_TLS int _signals_masked = 0;
```

Comment:
ZEND_API TSRM_TLS is on pretty thin ice... I assume this code doesn't apply to Windows? In that case it might work...

Reply:
No, it doesn't apply to Windows. `ZEND_TLS` makes the declaration `static`, which I didn't want, since detection of (incorrect) re-entrant use would not apply across source files.

But maybe it doesn't matter!

Reply:
Oh right, it's even static. In that case the ZEND_API certainly doesn't make sense :)

I'd suggest to just add two exported functions to do the increment/decrement. As these are only used for debug builds, it shouldn't matter.

Reply:
@nikic, I'm interested in understanding why you say that "TSRM_TLS is on thin ice".

Reply:
Specifically the combination of ZEND_API (shared object export) and TLS is problematic, because Windows does not support cross-DLL thread local storage.

That's basically the reason why we still have the whole TSRM machinery, because we have many thread-safe globals shared across DLLs/SOs. So it's necessary to either place it in a TSRM global, or go through functions to access the value.

Reply:
OK, just thinking this through... it's starting to make sense...

Reply:
OK, please have a look now.